db/crud.go

```diff
@@ -187,6 +187,31 @@ func (c *DatabaseCollection) GetDocSyncData(ctx context.Context, docid string) (
 
 }
 
+// This gets *just* the Sync Metadata (_sync field) rather than the entire doc, for efficiency
+// reasons. Unlike GetDocSyncData it does not check for on-demand import; this means it does not
+// need to read the doc body from the bucket.
+func (db *DatabaseCollection) GetDocSyncDataNoImport(ctx context.Context, docid string, level DocumentUnmarshalLevel) (SyncData, error) {
+	if db.UseXattrs() {
+		var err error
+		if key := realDocID(docid); key == "" {
+			err = base.HTTPErrorf(400, "Invalid doc ID")
+		} else {
+			var cas uint64
+			var xattrValue []byte
+			if cas, err = db.dataStore.GetXattr(key, base.SyncXattrName, &xattrValue); err == nil {
+				var doc *Document
+				doc, err = unmarshalDocumentWithXattr(key, nil, xattrValue, nil, cas, level)
+				if err == nil {
+					return doc.SyncData, nil
+				}
+			}
+		}
+		return SyncData{}, err
+	} else {
+		return db.GetDocSyncData(ctx, docid)
```

Comment:
Is switching from the previous`GetDocument` to `GetDocSyncData` in the non-xattr case safe/equivalent?  GetDocument looks like it includes things like checkForUpgrade at least.

Reply:
Hm. Then shouldn't GetDocSyncData itself check for upgrade?

Reply:
Looks like GetDocSyncData is only called by the changes feed; maybe that isn't supposed to upgrade?
I'll add the checkForUpgrade code (from GetDocumentWithRaw) to GetDocSyncDataNoImport.